app/assets/src/components/views/SampleUploadFlow/UploadProgressModal.jsx

```diff
@@ -0,0 +1,486 @@
+    </React.Fragment>
+  );
+
+  renderTitle = () => {
+    const { samples, project, uploadType } = this.props;
+    const { uploadComplete, failedSampleNames } = this.state;
+
+    if (uploadType === "remote" || uploadType === "basespace") {
+      if (uploadComplete) {
+        // If any samples failed.
+        if (failedSampleNames.length > 0) {
+          return this.renderFailedSamplesTitle();
+          // If all samples succeeded.
+        } else {
+          return (
+            <React.Fragment>
+              <div className={cs.titleWithIcon}>
+                <CheckmarkIcon className={cs.checkmarkIcon} />
+                {samples.length} samples successfully created.
+              </div>
+              <div className={cs.instructions}>
+                Our servers have started uploading your sample files from{" "}
+                {uploadType === "basespace" ? "Basespace" : "S3"}. After the
+                upload is complete, your samples will automatically start
+                processing. Please check back soon.
+              </div>
+            </React.Fragment>
+          );
+        }
+      }
+      // While samples are still being created.
+      return (
+        <React.Fragment>
+          <div className={cs.title}>
+            Creating {samples.length} sample{samples.length !== 1 && "s"} to{" "}
+            {project.name}
+          </div>
+          <div className={cs.subtitle}>Please do not leave this page.</div>
+        </React.Fragment>
+      );
+    }
+
+    // While local samples are being uploaded.
+    if (this.someLocalSamplesInProgress()) {
+      return (
+        <React.Fragment>
+          <div className={cs.title}>
+            Uploading {samples.length} sample{samples.length !== 1 && "s"} to{" "}
+            {project.name}
+          </div>
+          <div className={cs.subtitle}>
+            Please stay on this page until your upload completes.
+          </div>
+        </React.Fragment>
+      );
+    }
+
+    // If any local samples failed.
+    if (this.someLocalSamplesFailed()) {
+      return this.renderFailedSamplesTitle();
+    }
+
+    // If all local samples succeeded.
+    return (
+      <React.Fragment>
+        <div className={cs.titleWithIcon}>
+          <CheckmarkIcon className={cs.checkmarkIcon} />
+          All samples uploaded successfully!
+        </div>
+      </React.Fragment>
+    );
+  };
+
+  renderSampleLoadingBar = sample => {
+    const uploadPercentage = this.getSampleUploadPercentage(sample);
+
+    return (
+      <div className={cs.loadingBarBackground}>
+        <div
+          className={cs.loadingBar}
+          style={{ width: `${uploadPercentage * 100}%` }}
+        />
+      </div>
+    );
+  };
+
+  renderViewProjectButton = () => {
+    const { project } = this.props;
+    const { uploadComplete } = this.state;
+    if (!uploadComplete) {
+      return null;
+    }
+
+    return (
+      <a className={cs.link} href={`/home?project_id=${project.id}`}>
```

Comment:
There doesn't seem to be an `href` prop on the PrimaryButton.

I could do something like `window.location = url` on the button. But I think `<a>` is preferred, because then you get all the browser support around hyperlinks (the link is shown when hovered, you can right click on the link target and get additional options, etc.)

I think the thing that's weird here is that the `<Button>` isn't doing anything other than sending off a logging event. It's more here for the styling. It's a little weird, but I don't think there's anything wrong with it as long as we're okay using buttons that way.

Reply:
There is not href prop, but if we want this effect I think we should add it on the button. Should be easy to extend it with an href prop.

Reply:
Anyway, we could event discuss that, since it is actually a link, it should look like a link...
Will let you decide on this.